model/fn-execution/src/main/proto/beam_provision_api.proto

```diff
@@ -58,4 +58,42 @@ message ProvisionInfo {
     // (required) Pipeline options. For non-template jobs, the options are
     // identical to what is passed to job submission.
     google.protobuf.Struct pipeline_options = 3;
+
+    // (optional) Resource limits that the SDK harness worker should respect.
+    // Runners may -- but are not required to -- enforce any limits provided.
+    Resources resource_limits = 4;
+}
+
+// Resources specify limits for local resources, such memory and cpu. It
+// is used to inform SDK harnesses of their allocated footprint.
+message Resources {
+    // Memory limits.
+    message Memory {
+        // (optional) Hard limit in bytes. A zero value means unspecified.
```

Comment:
Right

Reply:
Btw, the comment states the other direction: if you explicitly set it to zero, we'll still interpret the value as "unspecified", because using no memory is not meaningful.

Reply:
What I mean is that it is confusing and maybe misleading to describe this as optional. The field always has a value.

Reply:
That is true for all proto3 fields. I understand optional as "you don't have to set this value" and, if not obvious, one should specify the interpretation of the default value. What do you suggest instead?